pkg/ingest/ingest.go

```diff
@@ -0,0 +1,101 @@
+package ingest
+
+import (
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+
+	"github.com/thanos-community/obslytics/pkg/dataframe"
+	"github.com/thanos-community/obslytics/pkg/output"
+)
+
+type Ingestor interface {
+	Ingest(*storepb.Series) error
+}
+
+// Options for a single aggregation.
+type AggrOption struct {
+	// Should the aggregation be used?
+	Enabled bool
+	// Column to store the aggregation at
+	Column string
+}
+
+// Collections of aggregations-related options. Determines what aggregations are enabled etc..
+type aggrsOptions struct {
+	// If true (default), the samples should repeat the same time for every day.
+	// This puts additional requirement for 24h to be divisible by the resolution.
+	MidnightAligned bool
+
+	Sum   AggrOption
+	Count AggrOption
```

Comment:
I wonder what we do with counters  and rates. I think we need to have some `rate` aggregation? :thinking: WDYT? We can start with this

Reply:
I think it would make sense, it's just logic I'm not familiar with too much right now, so leaving it up to more contributions :)